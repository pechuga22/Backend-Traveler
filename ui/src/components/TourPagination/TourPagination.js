import React from 'react'
import InfiniteScroll from 'react-infinite-scroll-component'
import { Grid, Header, Icon, Container } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import TourCard from '../TourPackages/TourCard'
import Loading from '../Loading/Loading'

export default function TourPagination({ tours, getAllTours, loading }) {
  return (
    <Container text>
      <Grid columns={3} padded>
        <Grid.Column width={4}>
          <Link to={'/'}>
            <Icon color='pink' size='large' className='arrow left header' />
          </Link>
        </Grid.Column>
        <Grid.Column textAlign='center' width={8}>
          <Header color='pink'>
            <h2>See All Tours</h2>
          </Header>
        </Grid.Column>
      </Grid>
      <Grid.Row>
        <InfiniteScroll
          dataLength={tours.length}
          next={getAllTours}
          hasMore={true}
          loader={loading && <Loading />}
          style={{ overflow: 'hidden' }}
        >
          <Grid columns={2}>
            {tours &&
              tours.map((tour) => <TourCard tour={tour} key={tour._id} />)}
          </Grid>
        </InfiniteScroll>
      </Grid.Row>
    </Container>
  )
}
