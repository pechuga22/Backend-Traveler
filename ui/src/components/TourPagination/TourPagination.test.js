import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TourPagination from './TourPagination'

const tours = [
  {
    _id: '5ec573d6f235707135952a94',
    title: 'El mejor recorrido en Bici hasta Incachaca',
  },
  {
    _id: '5ec573d6f235707135952a95',
    title: 'Camino a la piedra de Guatapé',
  },
  {
    _id: '5ec573d6f235707135952a96',
    title: 'Historia de las playas de Bahia a pie',
  },
  {
    _id: '5ec574f6f235707135952a97',
    title: 'Cuba en bus, conociendo la Revolución Cubana',
  },
  {
    _id: '5ec574f6f235707135952a98',
    title: 'Ruta del tintico, lo mejor del café de Colombia',
  },
  {
    _id: '5ec574f6f235707135952a99',
    title: 'Tour de Boliches en la Ciudad Maravilla',
  },
]

describe('./components/TourPagination/TourPagination.js', () => {
  describe('.render()', () => {
    it('should render TourPagination component', () => {
      const wrapper = shallow(
        <TourPagination tours={tours} returnPath={'/alltours'} />
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders text TourPagination', () => {
      const wrapper = shallow(
        <TourPagination tours={tours} returnPath={'/alltours'} />
      )
      expect(wrapper.find('h2').text()).toBe('See All Tours')
    })
  })
})
