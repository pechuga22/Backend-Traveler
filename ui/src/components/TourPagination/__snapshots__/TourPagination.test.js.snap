// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/TourPagination/TourPagination.js .render() should render TourPagination component 1`] = `
<Container
  text={true}
>
  <Grid
    columns={3}
    padded={true}
  >
    <GridColumn
      width={4}
    >
      <Link
        to="/"
      >
        <Icon
          as="i"
          className="arrow left header"
          color="pink"
          size="large"
        />
      </Link>
    </GridColumn>
    <GridColumn
      textAlign="center"
      width={8}
    >
      <Header
        color="pink"
      >
        <h2>
          See All Tours
        </h2>
      </Header>
    </GridColumn>
  </Grid>
  <GridRow>
    <InfiniteScroll
      dataLength={6}
      hasMore={true}
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <Grid
        columns={2}
      >
        <TourCard
          key="5ec573d6f235707135952a94"
          tour={
            Object {
              "_id": "5ec573d6f235707135952a94",
              "title": "El mejor recorrido en Bici hasta Incachaca",
            }
          }
        />
        <TourCard
          key="5ec573d6f235707135952a95"
          tour={
            Object {
              "_id": "5ec573d6f235707135952a95",
              "title": "Camino a la piedra de Guatapé",
            }
          }
        />
        <TourCard
          key="5ec573d6f235707135952a96"
          tour={
            Object {
              "_id": "5ec573d6f235707135952a96",
              "title": "Historia de las playas de Bahia a pie",
            }
          }
        />
        <TourCard
          key="5ec574f6f235707135952a97"
          tour={
            Object {
              "_id": "5ec574f6f235707135952a97",
              "title": "Cuba en bus, conociendo la Revolución Cubana",
            }
          }
        />
        <TourCard
          key="5ec574f6f235707135952a98"
          tour={
            Object {
              "_id": "5ec574f6f235707135952a98",
              "title": "Ruta del tintico, lo mejor del café de Colombia",
            }
          }
        />
        <TourCard
          key="5ec574f6f235707135952a99"
          tour={
            Object {
              "_id": "5ec574f6f235707135952a99",
              "title": "Tour de Boliches en la Ciudad Maravilla",
            }
          }
        />
      </Grid>
    </InfiniteScroll>
  </GridRow>
</Container>
`;
